aten/src/ATen/core/Device.cpp

```diff
@@ -64,7 +64,11 @@ Device::Device(const std::string& device_string) : Device(Type::CPU) {
 
   const auto index_string = device_string.substr(position);
   try {
+#if defined(__ANDROID__)
+    index_ = stoi(index_string);
```

Comment:
is this how ANDROID-specific calls like this are handled in the caffe2 codebase?  It seems a bit strange because you'd have to guard every call site this way rather than have a single implementation that does the right thing, but I don't know much about mobile.